src/test/java/com/xebialabs/xlt/ci/server/XLTestServerImplTest.java

```diff
@@ -98,30 +126,97 @@ public void fillsTestSpecificationIdItems() {
 
     @Test
     public void shouldImport() throws Exception {
-        xltStub.enqueue(new MockResponse()
+        xltestviewMock.enqueue(new MockResponse()
                 .addHeader("Content-Type", "application/json; charset=utf-8")
                 .setBody("{ \"testRunId\": \"testrunid\" }"));
-        FilePath fp = new FilePath(new File("."));
+        FilePath fp = new FilePath(new File(this.getClass().getResource("/demo_test_results").getPath()));
 
         PrintStream logger = new PrintStream(System.out);
 
-        Map<String, Object> metadata = new LinkedHashMap<String, Object>();
-        metadata.put("source", "jenkins");
-        metadata.put("serverUrl", "http://my-jenkins");
-        metadata.put("buildResult", "success");
-        metadata.put("buildNumber", "10");
-        metadata.put("jobUrl", "http://my-jenkins/job/sub-dir/my-job");
-        metadata.put("jobName", "sub-dir/my-job");
-        metadata.put("buildUrl", "http://my-jenkins/job/test/14");
-        metadata.put("executedOn", "slave1");
-        metadata.put("buildParameters", new HashMap());
+        Map<String, Object> metadata = createMetadata();
 
         xlTestServer.uploadTestRun("testspecid", fp, "**/*.xml", null, metadata, logger);
 
-        RecordedRequest request = xltStub.takeRequest();
+        RecordedRequest request = xltestviewMock.takeRequest();
+        verifyUploadRequest(request);
+    }
+
+    @Test
+    public void shouldImportWithTrailingSlash() throws IOException, InterruptedException, MessagingException {
```

Comment:
I do not agree. We test the usage of the JVM class, not the JVM class itself. shouldImportWithTrailingSlash tests what happens if somebody types in a trailing slash, and verifies that the call is still what it should be, without a double slash. This is exactly the scenario that went wrong, and what we want to test